### ISSUE TITLE
Night Mode

### ISSUE THREAD

**Issue by DeltaManiac** [rust-community/rust-lang-blog-boilerplate#3]
Is there a night mode/dark variant?

**Comment by o0Ignition0o** [comment]
Hi and welcome to the project :)
There isn't any night mode yet, but that sounds like a good idea.
If you find one or create your own, please let me know !

**Comment by DeltaManiac** [comment]
I dont know much about CSS but I Fiddled around with the [ayu-rs](https://github.com/Cldfire/ayu-rs/blob/master/css/twir.css) theme for this-week-in-rust and it supports night mode for this blog template with a small change.
```css
.highlight {
    background-color:#191f26
}

pre,code {
        background-color:#191f26
}

#site-logo {
        filter: invert(100%);
}
```

Just stick the above code to the end and Voila!😎

![Landing Page](https://user-images.githubusercontent.com/3689561/42558203-2c4aef80-850e-11e8-9874-aab26c3b0b7b.png)


![Post](https://user-images.githubusercontent.com/3689561/42558176-123a94ec-850e-11e8-8e02-497011c05c38.png)

**Comment by o0Ignition0o** [comment]
Wow this looks great ! congratulations !

AFAIR there's a way to toggle a theme using Jekyll, so we might want to call this one the night-theme or something, and make it configurable.

Anyway it looks awesome, I might use it in my own blog :D

**Comment by Cldfire** [comment]
Hello! Ayu-rs author here. I'd love to put together a slightly modified version of the theme for use here. Would you (@o0Ignition0o) be willing to explain a bit more about how to add a configurable theme so I can get started on a PR? Or, alternatively, I can get the CSS ready and then hand it over to you to integrate. Let me know what works better for you 😃

**Comment by o0Ignition0o** [comment]
Heya !

I really love you all pushing new themes and stuff, you are awesome <3 

I'm gonna take a bit of time tomorrow and figure out how to enable a theme switch flag, which isn't supposed to be too hard on Jekyll.

I'll let you know how it goes.

Again thank y'all soooo much you're awesome! <3

**Comment by DeltaManiac** [comment]
@o0Ignition0o 
[ayu-rs](https://github.com/Cldfire/ayu-rs/)  as of [#17](https://github.com/Cldfire/ayu-rs/pull/17) supports dark mode with more consistent theme for night for twir/rust-blog.

**Comment by o0Ignition0o** [comment]
From what I see [here](https://stackoverflow.com/questions/31327045/switch-theme-in-an-existing-jekyll-installation) there might be a way to switch themes but it seems a bit of a struggle, I probably need to dig deeper

**Comment by DeltaManiac** [comment]
@o0Ignition0o 

I was able to get night mode working for the blog via a configuration in the config.yml.
[Night Mode via config.yml](https://github.com/DeltaManiac/rust-lang-blog-boilerplate/commit/c1ffd0f9c40de5fa480f8a486ec53ca962cf0729)

Do you have any thoughts on how we could make this configurable via the UI ? 

One thought on the top of my head is to toggle the class via javascript and storing the preferred them in a cookie perhaps.

**Comment by o0Ignition0o** [comment]
Hi @DeltaManiac , 

This sounds great to me :)

We might want to consider people having Javascript disabled on their browser and maybe keep the light mode if it's the case ?  
Afair cookies are for client / server stateful communication, so we might consider localstorage instead of cookies? 

Edit : https://stackoverflow.com/questions/3220660/local-storage-vs-cookies seems to go with that ?

**Comment by DeltaManiac** [comment]
Hey @o0Ignition0o 

Thanks for the link, i got it working with local storage :) 
![night_mode](https://user-images.githubusercontent.com/3689561/43465183-74e451e0-94fa-11e8-9070-2223da8b1eda.gif)
The button looks ugly because of me lacking CSS skills, if anyone could make it look better that'd be rad.

**Comment by o0Ignition0o** [comment]
Wow it looks amazing !
I'm no ui expert myself, but I can have a look at it tonight :) I was thinking of a day / night switch or something

**Comment by DeltaManiac** [comment]
Awesome.
May be @Cldfire could also take a look at the NightMode CSS because its the same ayu-rs theme

**Comment by Cldfire** [comment]
I won't be able to look at anything until the week after next, but I'd be happy to do so then 👍

**Comment by o0Ignition0o** [comment]
just found a cool button on the docker documentation site : 
![image](https://user-images.githubusercontent.com/9465678/44195894-b4e6cd80-a13a-11e8-8eba-b1706c675edd.png)

https://docs.docker.com/docker-cloud/apps/service-links/#service-environment-variables for reference (on the right hand side)

I think it would look great, what do you think ?  :D

**Comment by jasonwilliams** [comment]
> just found a cool button on the docker documentation site :
> ![image](https://user-images.githubusercontent.com/9465678/44195894-b4e6cd80-a13a-11e8-8eba-b1706c675edd.png)
> 
> https://docs.docker.com/docker-cloud/apps/service-links/#service-environment-variables for reference (on the right hand side)
> 
> I think it would look great, what do you think ? :D

I've had a go today, ignore the toggle on the top

![toggle](https://user-images.githubusercontent.com/936006/85955965-1e140a80-b97a-11ea-9260-49f9b528ab06.gif)

**Comment by o0Ignition0o** [comment]
It looks really great!

**Comment by o0Ignition0o** [comment]
Fixed through #4 and #11 :tada:
